### PR TITLE
Updated reactor core, reactor netty and netty versions to latest

### PR DESCRIPTION
**Pull request by kushagraThapar**
* Updated reactor core to `3.3.8.RELEASE` - release notes : https://github.com/reactor/reactor-core/releases/tag/v3.3.8.RELEASE
* Updated reactor-netty to `0.9.10.RELEASE` - release notes : https://github.com/reactor/reactor-netty/releases/tag/v0.9.10.RELEASE
* Updated netty to `4.1.51.Final` - release notes : https://netty.io/news/2020/07/09/4-1-51-Final.html
* Updated netty tcnative to `2.0.31.Final` - release notes : https://github.com/netty/netty-tcnative/releases/tag/netty-tcnative-parent-2.0.31.Final
* Updated netty tcnative boring ssl to `2.0.31.Final`

* The reason for these updates is the reactor-netty issue that azure-cosmos is facing : https://github.com/reactor/reactor-netty/issues/1165

* This fixes issue : https://github.com/Azure/azure-sdk-for-java/issues/7349

### CODE REVIEW

File: eng/versioning/external_dependencies.txt [hunk 1]
```diff
@@ -32,18 +32,18 @@ commons-codec:commons-codec;1.13
...
-io.netty:netty-codec-http2;4.1.49.Final
-io.netty:netty-handler;4.1.49.Final
-io.netty:netty-handler-proxy;4.1.49.Final
+io.netty:netty-buffer;4.1.51.Final
+io.netty:netty-codec-http;4.1.51.Final
+io.netty:netty-codec-http2;4.1.51.Final
+io.netty:netty-handler;4.1.51.Final
+io.netty:netty-handler-proxy;4.1.51.Final
 io.netty:netty-tcnative;2.0.29.Final
 io.netty:netty-tcnative-boringssl-static;2.0.29.Final
```
Comment (alzimmermsft) [inline eng/versioning/external_dependencies.txt:41]: Looks like the newer versions of `netty-handler` use version `2.0.31` of these artifacts.

Comment (kushagraThapar) [reply eng/versioning/external_dependencies.txt:41]: good catch @alzimmermsft - I will update those as well.

File: sdk/core/azure-core/pom.xml [hunk 2]
```diff
@@ -84,7 +84,7 @@
     <dependency>
       <groupId>io.projectreactor</groupId>
       <artifactId>reactor-core</artifactId>
-      <version>3.3.5.RELEASE</version> <!-- {x-version-update;io.projectreactor:reactor-core;external_dependency} -->
+      <version>3.3.8.RELEASE</version> <!-- {x-version-update;io.projectreactor:reactor-core;external_dependency} -->
```
Comment (srnagar) [inline sdk/core/azure-core/pom.xml:87]: Could you please update the changelog and call out the version update for `azure-core`, `azure-core-http-netty` and `azure-cosmos` modules?

Comment (kushagraThapar) [reply sdk/core/azure-core/pom.xml:87]: @srnagar - sure will do that once the cosmos live test CI finishes.

Comment (kushagraThapar) [reply sdk/core/azure-core/pom.xml:87]: @srnagar - done, please check.

**Pull request merged**